audio/src/lib.rs

```diff
@@ -29,8 +30,70 @@ pub use fetch::{
     READ_AHEAD_BEFORE_PLAYBACK_ROUNDTRIPS, READ_AHEAD_BEFORE_PLAYBACK_SECONDS,
     READ_AHEAD_DURING_PLAYBACK_ROUNDTRIPS, READ_AHEAD_DURING_PLAYBACK_SECONDS,
 };
+use std::fmt;
+
+pub enum AudioPacket {
+    Samples(Vec<i16>),
+    OggData(Vec<u8>),
+}
+
+impl AudioPacket {
+    pub fn samples(&self) -> &[i16] {
+        match self {
+            AudioPacket::Samples(s) => s,
+            AudioPacket::OggData(_) => panic!("can't return OggData on samples"),
+        }
+    }
+
+    pub fn oggdata(&self) -> &[u8] {
+        match self {
+            AudioPacket::Samples(_) => panic!("can't return samples on OggData"),
+            AudioPacket::OggData(d) => d,
+        }
+    }
+
+    pub fn is_empty(&self) -> bool {
+        match self {
+            AudioPacket::Samples(s) => s.is_empty(),
+            AudioPacket::OggData(d) => d.is_empty(),
+        }
+    }
+}
 
 #[cfg(not(any(feature = "with-tremor", feature = "with-vorbis")))]
-pub use crate::lewton_decoder::{VorbisDecoder, VorbisError, VorbisPacket};
+pub use crate::lewton_decoder::{VorbisDecoder, VorbisError};
 #[cfg(any(feature = "with-tremor", feature = "with-vorbis"))]
-pub use libvorbis_decoder::{VorbisDecoder, VorbisError, VorbisPacket};
+pub use libvorbis_decoder::{VorbisDecoder, VorbisError};
+pub use passthrough_decoder::{PassthroughDecoder, PassthroughError};
+
+#[derive(Debug)]
+pub enum AudioError {
+    PassthroughError(PassthroughError),
+    VorbisError(VorbisError),
+}
```

Comment:
Is it the right way to make `AudioDecoder` a trait, but `AudioError` an enum?

Reply:
done then